### ISSUE TITLE
Erroneous results?

### ISSUE THREAD

**Issue by ivan-marroquin** [ialuronico/RandomizedInformationCoefficient#2]
Hi,

I am using octave 5.1.0 on a windows machine. To test RIC, I generated two sets of 2D synthetic data: linear and nonlinear distributions, following the Python code at https://minepy.readthedocs.io/en/latest/python.html

Then, I compile RIC using the compile_RIC_matlab script without issues.

I created my own script to compute RIC on generated data sets and got these results:
A) all linear distributions, RIC correlation of 0
B) all  nonlinear distributions, RIC correlation of 0

Do these results make sense?

I attached to this issue: my script and input data.

Many thanks,
Ivan

[Matlab_RIC_2D_Synthetic_Data.zip](https://github.com/ialuronico/RandomizedInformationCoefficient/files/3135206/Matlab_RIC_2D_Synthetic_Data.zip)

**Comment by simoroma** [comment]
I am really sorry for the late reply. I didn't receive any notification for this issue. Probably because I lost the credentials for this account.

Anyway I just checked your code. As it is now implemented, RIC requires to input row vectors. Change e.g. `Statistics(count,1)= RIC(attributes_data(:,i), attributes_data(:,(i+1)));` into `Statistics(count,1)= RIC(attributes_data(:,i)', attributes_data(:,(i+1))');`.

This snipped would work:
```
     x = rand(1,100)
     y = rand(1,100)
     RIC(x, x)
     RIC(x, y)
```
